### PR TITLE
Estimate burn gas fee

### PR DESCRIPTION
**Pull request by dweberdev**
Estimate burn gas fee

### CODE REVIEW

File: archetypes/Exchange/Summary/index.tsx [hunk 1]
```diff
@@ -40,15 +40,16 @@ export default (({ pool, showBreakdown, amount, isLong, receiveIn, inputAmount,
...
-        if (amount.toNumber() <= 0) {
+    const getCommitGasFee = (action: string) => {
+        if (commitAction === action && amount.toNumber() === 0) {
             return 0;
-        } else if (mintGasFee < 0.001) {
+        } else if (commitGasFees[action] < 0.001) {
             return '< $0.001';
         } else {
-            return toApproxCurrency(mintGasFee.toFixed(3));
+            return toApproxCurrency(commitGasFees[action]?.toFixed(3));
```
Comment (dospore) [inline archetypes/Exchange/Summary/index.tsx:52]: I think we could just pass the appropriate gas fee in props

Comment (dospore) [reply archetypes/Exchange/Summary/index.tsx:52]: that way we can avoid the 'any' props as well

File: archetypes/Exchange/index.tsx [hunk 2]
```diff
@@ -31,10 +31,16 @@ const TRADE_OPTIONS = [
     },
 ];
 
+type CommitGasTypes = {
+    mint?: 'string';
+    burn?: 'string';
+    flip?: 'string';
+};
```
Comment (dospore) [inline archetypes/Exchange/index.tsx:38]: we can make this type Partial<Record<CommitActionEnum, string>>, and pass the commit gas fee in with 

gasFee = {commitGasFees[commitAction]}

Comment (dweberdev) [reply archetypes/Exchange/index.tsx]: Not sure I implemented correctly but I kept getting TS errors when trying to define the state type like above.

Passing prop was a bit of struggle.. ended up having to do this:
gasFee={commitGasFees[CommitActionEnum[commitAction] as keyof typeof CommitActionEnum]}

**Review by dospore — APPROVED** [review]
lgtm

**Review by dospore — APPROVED** [review]

**Pull request merged**